src/templates_front/build.sh

```diff
@@ -25,9 +25,9 @@ cat prefix_python.py   > trexio.py
 # append version string and attributes to the Fortran module file
 echo "" >> trexio_f.f90
 echo "character(len = 12) :: TREXIO_PACKAGE_VERSION = ${VERSION_VAL}" >> trexio_f.f90
-echo "integer(4) :: TREXIO_VERSION_MAJOR = ${VERSION_MAJOR_VAL}" >> trexio_f.f90
-echo "integer(4) :: TREXIO_VERSION_MINOR = ${VERSION_MINOR_VAL}" >> trexio_f.f90
-echo "integer(4) :: TREXIO_VERSION_PATCH = ${VERSION_PATCH_VAL}" >> trexio_f.f90
+echo "integer(c_int32_t) :: TREXIO_VERSION_MAJOR = ${VERSION_MAJOR_VAL}" >> trexio_f.f90
```

Comment:
I am not sure that these tangled versions have to be `c_int32_t` since they have nothing to do with ISO_C_BINDING. These constants are intended to be used internally in Fortran programs, the C API calls the same constants that are declared in `trexio.h`.

Reply:
I agree it is not necessary technically, but this is because:
1. `integer(4)` is not in the Fortran norm, although it is accepted by most (if not all!) compilers.
2. I don't want to use the `selected_kind` syntax because I hate it
3. It is a way to specify that you always want 32 bits, whatever the architecture.
4. It avoids mixing different syntaxes

Reply:
I do not think that the version numbers of TREXIO will ever exceed the limit of the 32-bit integer, or at least I hope they will not :-) 

My point was that we could probably use just `integer` without losing anything here.

Reply:
Yes.

Reply:
So do you want to keep `integer(c_int32_t) :: ` or we agree on using `integer :: ` for these constants ?

Reply:
OK, so we agreed on this. I made the necessary changes in #734f813544155783f013aa33bbf1a3c986cf2206 .